### ISSUE TITLE
Switсh - is considered a lamp and reduces their maximum allowable quantity

### ISSUE THREAD

**Issue by MaaiG** [Regalis11/Barotrauma#9435]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

In addition, despite the lack of reaction to the change of its light settings, the switch can also will be a shadow casting lights!

Given the current limit on lamps, this situation could be considered wasteful.
I love glow buttons, but I don’t want them to be considered lamps, let alone shadow casting...

Is that not a mistake?
Is it possible to remove them from the lighting limit?
https://user-images.githubusercontent.com/93348499/174651770-38521d5e-2f35-47de-a6f4-eb15b7f9f6f6.mp4

### Reproduction steps

1. Editor
2. Create switch
3. Look at the number of lights in the window "Total quantity objects

### Bug prevalence

Happens every time I play

### Version

0.18.12.0

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Yes, the switches have light components (which do have a performance impact, even though very small), so they are taken to account in the total light count.

**Comment by MaaiG** [comment]
If 10 switches equals 1 lamp ...or something
then it is possible to count them accordingly 😉

**Comment by Ek-Videogames** [comment]
> Given the current limit on lamps, this situation could be considered wasteful. I love glow buttons, but I don’t want them to be considered lamps, let alone shadow casting... ![Image 001_20-27](https://github.com/assets/93348499/782b78fd-b006-4600-a67c-53e19e60392b)
> 
> Is that not a mistake? Is it possible to remove them from the lighting limit? https://user-images.githubusercontent.com/93348499/174651770-38521d5e-2f35-47de-a6f4-eb15b7f9f6f6.mp4

This is a longstanding known _fun_-quality of the editor that most sub builders are aware of,
IIRC basically as soon as the item counter was added to the GUI.

So because Switches/Junctions/Consoles/Suits/Tools it isn't really the best representation of actual sub light count.
There's usually worse performance offenders anyways as long as there's just one **"real"** light per room.

If the game were hypothetically on a different engine, then 'self-illumination' for sprites could be done cheap without problem, but due to how the lighting system works it's what we're stuck with.
